### PR TITLE
Render ts

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules/angular2/globals.ts [hunk 1]
```diff
@@ -20,4 +22,6 @@ interface Window {
   assert: typeof assert;
   gc(): void;
   Reflect: any;
+  zone: any;
+  Hammer: any;
```
Comment (alexeagle) [inline modules/angular2/globals.ts]: can we avoid any here?

File: modules/angular2/src/core/zone/ng_zone.ts [hunk 2]
```diff
@@ -19,12 +19,12 @@ export class NgZone {
   // onTurnDone hook at the end of the current VM turn.
   _innerZone;
 
-  _onTurnStart:Function;
-  _onTurnDone:Function;
-  _onErrorHandler:Function;
+  _onTurnStart: Function;
```
Comment (alexeagle) [inline modules/angular2/src/core/zone/ng_zone.ts]: instead of Function could be () => void; ?

File: modules/angular2/src/render/api.ts [hunk 3]
```diff
@@ -66,7 +70,7 @@ export class ElementBinder {
 
 export class DirectiveBinder {
   // Index into the array of directives in the View instance
-  directiveIndex:any;
+  directiveIndex: any;
```
Comment (alexeagle) [inline modules/angular2/src/render/api.ts]: is this any or number?

File: modules/angular2/src/render/api.ts [hunk 4]
```diff
@@ -111,25 +123,31 @@ export class ProtoViewDto {
...
+  hostActions: Map<string, string>;
+  properties: Map<string, string>;
+  readAttributes: List<string>;
+  type: number;
+  callOnDestroy: boolean;
+  callOnChange: boolean;
+  callOnAllChangesDone: boolean;
   changeDetection: string;
-  constructor({id, selector, compileChildren, events, hostListeners, hostProperties,
+  constructor({
```
Comment (alexeagle) [inline modules/angular2/src/render/api.ts]: we should think about how to do this without destructuring

File: modules/angular2/src/render/dom/compiler/compile_step.ts [hunk 5]
```diff
@@ -6,5 +6,6 @@ import * as compileControlModule from './compile_control';
  * Is guaranteed to be called in depth first order
  */
 export class CompileStep {
```
Comment (alexeagle) [inline modules/angular2/src/render/dom/compiler/compile_step.ts]: meant to be an interface?

File: modules/angular2/src/render/dom/compiler/selector.ts [hunk 6]
```diff
@@ -329,12 +338,12 @@ class SelectorListContext {
 
 // Store context to pass back selector and context when a selector is matched
 class SelectorContext {
-  selector:CssSelector;
-  notSelector:CssSelector;
-  cbContext; // callback context
+  selector: CssSelector;
+  notSelector: CssSelector;
+  cbContext;  // callback context
```
Comment (alexeagle) [inline modules/angular2/src/render/dom/compiler/selector.ts]: type?

File: modules/angular2/test/render/dom/dom_testbed.js [hunk 7]
```diff
@@ -10,13 +10,12 @@ import {RenderViewRef, ProtoViewDto, ViewDefinition, EventDispatcher, DirectiveM
 import {resolveInternalDomView} from 'angular2/src/render/dom/view/view';
 import {el, dispatchEvent} from 'angular2/test_lib';
 
-export class TestView extends EventDispatcher {
+export class TestView /* implements EventDispatcher */ {
```
Comment (alexeagle) [inline modules/angular2/test/render/dom/dom_testbed.js]: why commented?

Comment (tbosch) [inline modules/angular2/test/render/dom/dom_testbed.js]: As it is not typescript yet :-(

File: modules/angular2/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,8 @@
+{
+    "compilerOptions": {
+        "target": "es5",
```
Comment (rkirov) [inline modules/angular2/tsconfig.json]: add `emitDecoratorMetadata: true`

**Pull request merged**